From dd0cf46d98a6bd0b6ecf9ce1e77c672c5ddd4927 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Thu, 1 Dec 2011 21:59:12 +0800
Subject: [PATCH 178/280] ENGR00163617 vpu: Fix the incorrect color of NV12
 MJPG on mx6q

Fix the incorrect color of NV12 MJPG in decoder and encoder on mx6q.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_lib.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index fe795aa..003938e 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -1513,6 +1513,7 @@ RetCode vpu_EncStartOneFrame(EncHandle handle, EncParam * param)
 		while(!val)
 			val = VpuReadReg(GDI_STATUS);
 		VpuWriteReg(GDI_INFO_CONTROL, ((pEncInfo->jpgInfo.format & 0x07) << 17) |
+					       (pEncInfo->openParam.chromaInterleave << 16) |
 					       pSrcFrame->strideY);
 		VpuWriteReg(GDI_INFO_PIC_SIZE, (pEncInfo->jpgInfo.alignedWidth << 16) |
 						pEncInfo->jpgInfo.alignedHeight);
@@ -3552,6 +3553,7 @@ RetCode vpu_DecStartOneFrame(DecHandle handle, DecParam * param)
 			val = VpuReadReg(GDI_STATUS);
 
 		VpuWriteReg(GDI_INFO_CONTROL, ((pDecInfo->jpgInfo.format & 0x07) << 17) |
+					       (pDecInfo->openParam.chromaInterleave << 16) |
 						pDecInfo->rotatorStride);
 		VpuWriteReg(GDI_INFO_PIC_SIZE, (pDecInfo->jpgInfo.alignedWidth << 16) |
 						pDecInfo->jpgInfo.alignedHeight);
-- 
1.8.0

